builder/dockerfile/parser/parser.go

```diff
@@ -74,6 +93,26 @@ func parseLine(line string) (string, *Node, error) {
 		return "", nil, nil
 	}
 
+	// Handle the parser directive 'ESCAPE'. Parser directives must preceed
+	// any builder instruction, and cannot be repeated.
+	if tokenEscapeCommand.MatchString(strings.ToLower(line)) {
+		if builderInstructionProcessed == true {
```

Comment:
I'm not that bothered.... I deliberately didn't do a line 1 check in case future parser directives are added, any/all of which will (presumably) require being before any builder instructions.